docs/typescript/reference/dbos.md

```diff
@@ -0,0 +1,953 @@
+---
+sidebar_position: 4
+title: DBOS Class
+description: API reference for DBOS
+---
+
+# DBOS Class
+
+The `DBOS` class is the heart of the DBOS Transact programming API, providing decorators to identify key application functions, and accessors for context-dependent state.
+
+Note that this is the second major version of the API.  The first version of the API was based on passing [contexts](./contexts.md) and used a separate set of [decorators](./decorators.md).  It is fine to use both approaches in the same DBOS application, however the `DBOS` class is simpler and is therefore recommended.
+
+## Decorating Workflows, Transactions, and Steps
+DBOS provides reliability guarantees for programs that are [written as workflows comprised of steps, transactions, and child workflows](../programming-guide.md).  The workflow, transaction, and step functions are marked with decorators.
+
+### DBOS.workflow
```

Comment:
The styling is not consistent here. We can consistently use `@` in front of decorators, and make methods in `code blocks`. 
```suggestion
### `@DBOS.workflow`
```

Reply:
There are quite a number of these that I found on review, will do all in one commit.